src/compiler/emitter.ts

```diff
@@ -491,20 +497,120 @@ module ts {
                     }
                 }
 
+                function convertToBase64(input: string): string {
+                    // Encode a string into Base64 to include it in a URI for instance.
+                    // Effectively what we are doing here is putting every 3 charachters 
+                    // into 4 base64 digits.
+                    // UTF-8 makes this a bit tricky as each charater may need to be expanded
+                    // to 1, 2 or 3 charaters to ensure the conversion is correct.
+                    // To do that we consider one character from the input at a time, for non-ASCII 
+                    // character expand them into 2 or 3 as appropriate. Then take 3 expanded chars
+                    // at a time and convert them into 4 base-64 digits and add them to the ouput.
+
+                    // Output
+                    var result = "";
+
+                    // A working set of charachter points to write in base64 encoding. We only need
+                    // 3 character points at a time, so the most number of points we can have is 5.
+                    var chars: number[] = new Array(5);
+
+                    // The current index in the working set
+                    var charIndex = 0;
+
+                    var length = input.length;
+                    var i = 0;
+
+                    // 4 base64 digits
+                    let byte1: number, byte2: number, byte3: number, byte4: number;
+
+                    // Keep looping as long as we have more characters to process in the input
+                    // or character points to process in the working set.
+                    while (i < length || charIndex > 0) {
+                        while (i < length && charIndex < 3) {
+                            // This can be a utf-8 charater, expand it as appropriate

```

Comment:
I'm a bit confused too.  JavaScript strings aren't UTF8.  They're a sequence of 16bit codepoints.  i.e. each charCodeAt gives you a value <65536.  And you may need two codepoints (i.e. surrogates) to actually make a unicode character.

UTF8, on the other hand, using 1-4 bytes to encode a single characater.  

I'm not saying what you're doing is wrong.  Just that hte comment seems confusing and a bit misleading.